src/zb_links/api/link/links.py

```diff
@@ -163,60 +165,44 @@ def post(self):
         """Create a new link related to a zbMATH object"""
         args = request.args
 
-        zbl_val = args["zbl code"]
-        source_val = args["source identifier"]
-        source_name = args["partner name"]
+        de_val = args["DE number"]
+        source_val = args["external id"]
+        source_name = args["partner"]
         link_date = datetime.utcnow()
-        provider_id = 1
+        provider = helpers.get_provider()
 
         message_list = []
-        partner_name = None
 
         partner = Partner.query.filter_by(name=source_name).first()
         if partner:
-            partner_id = partner.partner_id
             partner_name = partner.name
         else:
             message_list.append("Invalid partner name")
-            partner_id = None
 
-        target_obj = ZBTarget.query.filter_by(zbl_code=zbl_val).first()
+        target_obj = ZBTarget.query.filter_by(id=de_val).first()
         if not target_obj:
-            message_list.append("Zbl code is not the database")
+            message_list.append("This DE is not in the database")
 
-        source_obj = Source.query.filter_by(
-            identifier=source_val, partner=partner_name
-        ).first()
-        if source_obj:
-            source_id = source_obj.source_id
-        else:
-            message_list.append("Invalid source identifier")
-            source_id = None
+        source_obj = Source.query.filter_by(id=source_val).first()
+        if not source_obj:
+            message_list.append("Invalid external id")
 
         if len(message_list) > 0:
             return helpers.make_message(422, message_list)
 
-        number_links = Link.query.count()
-
-        new_link_id = number_links + 1
-
+        new_link_id = link_helpers.get_new_id()
```

Comment:
This won't work. You can have multiple clients running queries at the same time. Thus they would get the same id. A naive approach would be using transactions. Without having looked into the details, I guess you want to return the new Primary Key https://stackoverflow.com/questions/42238903/sqlalchemy-return-generated-keys-for-insert

Reply:
@dehsani-academic have a look at
.inserted_primary_key
https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.ResultProxy.inserted_primary_key

Reply:
After looking again at your code, I see that you don't use this value at all. So the id can just be empty.

Reply:
exactly, the helper fcn should not be necessary.

Reply:
helper fcn has been erased